src/Provider/DatabaseServiceProvider.php

```diff
@@ -81,8 +85,14 @@ function ($app) {
             $app->extend(
                 'db',
                 function ($db) use ($app) {
+                    /** @var \Bolt\Storage\Database\Connection|\Bolt\Storage\Database\MasterSlaveConnection $db */
                     $db->setQueryCacheProfile($app['db.query_cache_profile']);
 
+                    // Currently, this can get called early, meaning the connection listener is too late
+                    if (DBAL\Version::compare('2.6.0') >= 0 && !Type::hasType('json')) {
```

Comment:
I'd drop the version comparison here. The feature detection is sufficient.

Reply:
I can't logic some days! Done :+1: